lib/openapi_first/ref_resolver.rb

```diff
@@ -55,7 +55,7 @@ def resolve_ref(pointer)
           value = Hana::Pointer.new(pointer[1..]).eval(context)
           raise "Unknown reference #{pointer} in #{context}" unless value
 
-          return RefResolver.for(value, dir:)
+          return RefResolver.for(value, dir:, context:)
```

Comment:
I think this change is correct. I also found this, but so far I was not able to reproduce the issue in a failing test inside ref_resolver_spec.rb

Reply:
Ok, I'll do my best to get a test in place to reproduce this. It's something I've been able to reproduce in my local project, so I think I have lead here.

Reply:
This is so hard to debug. I'd vote for removing the default values from RefResolver.for so we always have to specify context.

Reply:
>  I'd vote for removing the default values from RefResolver.for so we always have to specify context.

Tried this. It would require a lot of modifications to existing tests.